<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GraphBuildingForm</class>
 <widget class="QWidget" name="GraphBuildingForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>502</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="group_graph_builder">
     <property name="title">
      <string>Graph builder</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QTabWidget" name="tabs_graph_builder">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="tab_voxel_grid">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>279</width>
           <height>20</height>
          </size>
         </property>
         <attribute name="title">
          <string>Voxel Grid</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_voxel_resolution">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Voxel resolution</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="spinbox_voxel_resolution">
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.001000000000000</double>
            </property>
            <property name="maximum">
             <double>0.100000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.001000000000000</double>
            </property>
            <property name="value">
             <double>0.005000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_nearest_neighbors">
         <attribute name="title">
          <string>KNN</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_nearest_neighbors">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Nearest neighbors</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="spinbox_nearest_neighbors">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="value">
             <number>15</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="tab_radius">
         <attribute name="title">
          <string>Radius</string>
         </attribute>
         <layout class="QFormLayout" name="formLayout_2">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="4" column="1">
           <widget class="QDoubleSpinBox" name="spinbox_radius">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="decimals">
             <number>3</number>
            </property>
            <property name="minimum">
             <double>0.001000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>3.500000000000000</double>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_radius">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Radius</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QSpinBox" name="spinbox_max_neighbors">
            <property name="minimum">
             <number>2</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_max_neighbors">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Max neighbors</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="group_edge_weights">
     <property name="title">
      <string>Edge weights</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_8">
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_xyz_influence">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_xyz">
          <property name="text">
           <string>XYZ</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_xyz_only_concave">
          <property name="text">
           <string>Only concave</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_9">
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_normal_influence">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>0.005000000000000</double>
          </property>
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.005000000000000</double>
          </property>
          <property name="value">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_normal">
          <property name="text">
           <string>Normal</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_normal_only_concave">
          <property name="text">
           <string>Only concave</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_curvature_influence">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="minimum">
           <double>0.000100000000000</double>
          </property>
          <property name="maximum">
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.000100000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_curvature">
          <property name="text">
           <string>Curvature</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_curvature_only_concave">
          <property name="text">
           <string>Only concave</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_rgb_influence">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_rgb">
          <property name="text">
           <string>RGB</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_rgb_only_concave">
          <property name="text">
           <string>Only concave</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_21">
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_verticality_influence">
          <property name="decimals">
           <number>4</number>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>3.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_verticality">
          <property name="text">
           <string>Verticality</string>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="checkbox_verticality_only_concave">
          <property name="text">
           <string>Only concave</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_small_weight_threshold">
          <property name="text">
           <string>Small weight threshold:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="spinbox_small_weight_threshold">
          <property name="decimals">
           <number>6</number>
          </property>
          <property name="minimum">
           <double>0.000000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.000001000000000</double>
          </property>
          <property name="value">
           <double>0.000010000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_small_weight_policy">
          <property name="text">
           <string>Small weight policy:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="combo_small_weight_policy">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <item>
           <property name="text">
            <string>Ignore</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Coerce to threshold</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Remove</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="button_update">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>Update</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkbox_auto_update">
       <property name="text">
        <string>Auto update</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>checkbox_auto_update</sender>
   <signal>clicked(bool)</signal>
   <receiver>button_update</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>296</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>103</x>
     <y>356</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_radius</sender>
   <signal>editingFinished()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_max_neighbors</sender>
   <signal>editingFinished()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_nearest_neighbors</sender>
   <signal>editingFinished()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>261</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_voxel_resolution</sender>
   <signal>editingFinished()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>84</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkbox_xyz_only_concave</sender>
   <signal>clicked()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onEdgeWeightsEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>320</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabs_graph_builder</sender>
   <signal>currentChanged(int)</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkbox_xyz</sender>
   <signal>clicked()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onEdgeWeightsEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>199</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinbox_xyz_influence</sender>
   <signal>editingFinished()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onGraphBuilderEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>79</x>
     <y>215</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_update</sender>
   <signal>clicked()</signal>
   <receiver>GraphBuildingForm</receiver>
   <slot>onUpdateButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>356</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>188</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onGraphBuilderEditingFinished()</slot>
  <slot>onEdgeWeightsEditingFinished()</slot>
  <slot>onUpdateButtonClicked()</slot>
 </slots>
</ui>
